File: artifacts/ai-business-os/src/lib/business-industries.ts
export type BusinessIndustryGroup =
  | "agriculture"
  | "real_estate"
  | "commerce"
  | "healthcare"
  | "professional_services"
  | "other";

export type BusinessIndustryDefinition = {
  readonly label: string;
  readonly backendCode: string;
  readonly group: BusinessIndustryGroup;
  readonly isHealthcare: boolean;
  readonly supportsScheduling: boolean;
  readonly defaultTheme: "green" | "navy";
};

export const BUSINESS_INDUSTRY_DEFINITIONS = {
  "Farm/Agriculture": {
    label: "Farm/Agriculture",
    backendCode: "farm/agriculture",
    group: "agriculture",
    isHealthcare: false,
    supportsScheduling: false,
    defaultTheme: "green",
  },

  "Real Estate": {
    label: "Real Estate",
    backendCode: "real estate",
    group: "real_estate",
    isHealthcare: false,
    supportsScheduling: false,
    defaultTheme: "navy",
  },

  "E-commerce": {
    label: "E-commerce",
    backendCode: "e-commerce",
    group: "commerce",
    isHealthcare: false,
    supportsScheduling: false,
    defaultTheme: "green",
  },

  Hospital: {
    label: "Hospital",
    backendCode: "hospital",
    group: "healthcare",
    isHealthcare: true,
    supportsScheduling: true,
    defaultTheme: "green",
  },

  Clinic: {
    label: "Clinic",
    backendCode: "clinic",
    group: "healthcare",
    isHealthcare: true,
    supportsScheduling: true,
    defaultTheme: "green",
  },

  "Medical Practice": {
    label: "Medical Practice",
    backendCode: "medical practice",
    group: "healthcare",
    isHealthcare: true,
    supportsScheduling: true,
    defaultTheme: "green",
  },

  Dental: {
    label: "Dental",
    backendCode: "dental",
    group: "healthcare",
    isHealthcare: true,
    supportsScheduling: true,
    defaultTheme: "green",
  },

  "Professional Services": {
    label: "Professional Services",
    backendCode: "professional services",
    group: "professional_services",
    isHealthcare: false,
    supportsScheduling: true,
    defaultTheme: "green",
  },

  Other: {
    label: "Other",
    backendCode: "other",
    group: "other",
    isHealthcare: false,
    supportsScheduling: false,
    defaultTheme: "green",
  },
} as const satisfies Record<string, BusinessIndustryDefinition>;

export type OnboardingIndustry =
  keyof typeof BUSINESS_INDUSTRY_DEFINITIONS;

export const ONBOARDING_INDUSTRIES = Object.keys(
  BUSINESS_INDUSTRY_DEFINITIONS,
) as OnboardingIndustry[];

export const BUSINESS_INDUSTRY_BACKEND_ALIASES = {
  agriculture: "Farm/Agriculture",
  farm: "Farm/Agriculture",
  "farm/agriculture": "Farm/Agriculture",

  "real estate": "Real Estate",
  "real-estate": "Real Estate",

  ecommerce: "E-commerce",
  "e commerce": "E-commerce",
  "e-commerce": "E-commerce",

  hospital: "Hospital",
  clinic: "Clinic",
  medical: "Medical Practice",
  "medical practice": "Medical Practice",
  dental: "Dental",

  "professional service": "Professional Services",
  "professional services": "Professional Services",

  other: "Other",
} as const satisfies Record<string, OnboardingIndustry>;

export function normalizeBusinessIndustryBackendValue(
  value: string | null | undefined,
): string {
  return (value ?? "").trim().toLowerCase().replace(/\s+/g, " ");
}

export function businessIndustryLabelFromBackendCode(
  value: string | null | undefined,
): OnboardingIndustry | undefined {
  const normalized = normalizeBusinessIndustryBackendValue(value);

  if (!normalized) {
    return undefined;
  }

  const alias = BUSINESS_INDUSTRY_BACKEND_ALIASES[
    normalized as keyof typeof BUSINESS_INDUSTRY_BACKEND_ALIASES
  ];

  if (alias) {
    return alias;
  }

  return ONBOARDING_INDUSTRIES.find(
    (industry) =>
      BUSINESS_INDUSTRY_DEFINITIONS[industry].backendCode === normalized,
  );
}

export function getBusinessIndustryDefinition(
  industry: string | null | undefined,
): BusinessIndustryDefinition | undefined {
  if (!industry || !(industry in BUSINESS_INDUSTRY_DEFINITIONS)) {
    return undefined;
  }

  return BUSINESS_INDUSTRY_DEFINITIONS[
    industry as OnboardingIndustry
  ];
}

export function businessIndustrySupportsScheduling(
  industry: string | null | undefined,
): boolean {
  return getBusinessIndustryDefinition(industry)?.supportsScheduling ?? false;
}

export function isHealthcareIndustry(
  industry: string | null | undefined,
): boolean {
  return getBusinessIndustryDefinition(industry)?.isHealthcare ?? false;
}

export function businessIndustryBackendCode(
  industry: OnboardingIndustry,
): string {
  return BUSINESS_INDUSTRY_DEFINITIONS[industry].backendCode;
}

export function businessIndustryDefaultTheme(
  industry: string | null | undefined,
): "green" | "navy" {
  return getBusinessIndustryDefinition(industry)?.defaultTheme ?? "green";
}

File: backend/app/domain/business_industries.py
from __future__ import annotations

from dataclasses import dataclass
from types import MappingProxyType
from typing import Final, Mapping


@dataclass(frozen=True, slots=True)
class BusinessIndustryDefinition:
    """Server-owned metadata for a supported AI Business OS industry."""

    code: str
    label: str
    group: str
    is_healthcare: bool = False
    supports_scheduling: bool = False


_BUSINESS_INDUSTRIES = {
    "farm/agriculture": BusinessIndustryDefinition(
        code="farm/agriculture",
        label="Farm/Agriculture",
        group="agriculture",
    ),
    "real estate": BusinessIndustryDefinition(
        code="real estate",
        label="Real Estate",
        group="real_estate",
    ),
    "e-commerce": BusinessIndustryDefinition(
        code="e-commerce",
        label="E-commerce",
        group="commerce",
    ),
    "hospital": BusinessIndustryDefinition(
        code="hospital",
        label="Hospital",
        group="healthcare",
        is_healthcare=True,
        supports_scheduling=True,
    ),
    "clinic": BusinessIndustryDefinition(
        code="clinic",
        label="Clinic",
        group="healthcare",
        is_healthcare=True,
        supports_scheduling=True,
    ),
    "medical practice": BusinessIndustryDefinition(
        code="medical practice",
        label="Medical Practice",
        group="healthcare",
        is_healthcare=True,
        supports_scheduling=True,
    ),
    "dental": BusinessIndustryDefinition(
        code="dental",
        label="Dental",
        group="healthcare",
        is_healthcare=True,
        supports_scheduling=True,
    ),
    "professional services": BusinessIndustryDefinition(
        code="professional services",
        label="Professional Services",
        group="professional_services",
        supports_scheduling=True,
    ),
    "other": BusinessIndustryDefinition(
        code="other",
        label="Other",
        group="other",
    ),
}

BUSINESS_INDUSTRIES: Final[
    Mapping[str, BusinessIndustryDefinition]
] = MappingProxyType(_BUSINESS_INDUSTRIES)

BUSINESS_TYPE_ALIASES: Final[Mapping[str, str]] = MappingProxyType(
    {
        "agriculture": "farm/agriculture",
        "farm": "farm/agriculture",
        "farm/agriculture": "farm/agriculture",
        "real estate": "real estate",
        "real-estate": "real estate",
        "ecommerce": "e-commerce",
        "e commerce": "e-commerce",
        "e-commerce": "e-commerce",
        "hospital": "hospital",
        "clinic": "clinic",
        "medical": "medical practice",
        "medical practice": "medical practice",
        "dental": "dental",
        "professional services": "professional services",
        "professional service": "professional services",
        "other": "other",
    }
)

HEALTHCARE_BUSINESS_TYPES: Final[frozenset[str]] = frozenset(
    definition.code
    for definition in BUSINESS_INDUSTRIES.values()
    if definition.is_healthcare
)

SCHEDULING_BUSINESS_TYPES: Final[frozenset[str]] = frozenset(
    definition.code
    for definition in BUSINESS_INDUSTRIES.values()
    if definition.supports_scheduling
)


def normalize_business_type(value: str) -> str:
    """
    Normalize a business type while preserving forward compatibility.

    Known aliases resolve to their canonical server-owned code. Unknown values
    remain normalized strings rather than being rejected, because the existing
    Business domain intentionally supports bounded custom/future industries.
    """

    normalized = " ".join(value.strip().casefold().split())
    return BUSINESS_TYPE_ALIASES.get(normalized, normalized)


def get_business_industry(
    value: str,
) -> BusinessIndustryDefinition | None:
    return BUSINESS_INDUSTRIES.get(normalize_business_type(value))


def is_healthcare_business_type(value: str) -> bool:
    return normalize_business_type(value) in HEALTHCARE_BUSINESS_TYPES


def business_type_supports_scheduling(value: str) -> bool:
    return normalize_business_type(value) in SCHEDULING_BUSINESS_TYPES
